### DISCUSSION TITLE
Welcome to Cyb3rVector Discussions!

### DISCUSSION THREAD

**Discussion by cyb3rdog** [cyb3rdog/Cyb3rVector#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you think of the Cyb3rVector project.

**Comment by T4c03num** [comment]
Hey @cyb3rdog! First off, incredible tool you have developed!
I was curious as to a potential issue I am running into. So I am trying to create a simple behavior such as "Say hello world when you hear hello" or something along those lines. After I create the behavior and press play, it runs perfectly fine and I can see it on my escape pod as a new behavior generated by CyberVector. However, as soon as I hit stop that newly generated behavior is gone. Is this intended functionality or should it be writing newly created behaviors to vector and his escape pod?

**Reply by cyb3rdog** [reply]
Hey, thanks a lot for trying it out. I am super glad to hear somebody is using the Cyb3rVector EscapePod extension!

You are perfectly correct, for now, this is intended behaviour. Vector's "intents/behaviours", are generated by Cyb3rVector and are currently stored on EscapePod for only as long, as the actual Codelab code is running within the application itself... Once the codelab program is stopped, the record(s) is/are removed from the escapepod database.

Feature of deploying and exectuting the CodeLab code on the EscapePod itself, is however on the project's Roadmap. Once, I'll finish the "headless mode" for codelab programs, and the "task manager" milestones, you shall be able to deploy and manage the programs either on your local pc with cyb3rvector, or on the escapepod itself.

Many thanks for your support and stay tuned!

**Reply by starmindfr** [reply]
hello nice project i am Eric, a vector owner in france and this project is great. What i am looking for is mainly adding speechs in french for kids interaction as they dont speak english and vector voice is quite pain... I tried the current release and it's nice but need lots of love, by sample i try to collect the "voice" command sent to show it back on screen to understand what vector really ear, but most of time a crash of the application need to restart vector... maybe need some sort of error "try catch"... Made few attemps to capture voices commands and provide a list of text to read according too but i stop here as vector seems never to understand what i say, using the application and or skip it. Hope to see progress / feedback and will give few donation if all is still on the way

happy new year

**Reply by cyb3rdog** [reply]
Hi Eric, 

I have just written a reply to your issue. Please, read more about it [here](https://github.com/cyb3rdog/Cyb3rVector/issues/12).

Unfortunatelly, Vector's Voice recognition currently does not work the way you think. Production Vectors themselves, can only recognize the "Hey Vector" command, everything after that is sent over to DDL servers for voice transcription, and when the command is recognized, or a 'knowledge graph' / 'i have a question' answer is found, the so called 'intent' is sent back to Vector to perform something... From these intents, only few are currently passed further on through the SDK to the connected client applications.

In case you are a software developer with knowledge of Go, you may consider purchasing the OSKR and replacing the default voice recognition on DDL servers, with something of your own (this [DDL repository](https://github.com/digital-dream-labs/vector-cloud) might be good start for you).
The EscapePod can let you react to your own phrases, but it is perhaps not an option for you, as its embedded voice recognition is even little bit worse than on the DDL servers and is as well english only.

Learn more about the differences between [OSKR, EscapePod, and Production Vectors here](https://github.com/cyb3rdog/Cyb3rVector/discussions/6)

Luckily for us, DDL hopefully undestand these needs of ours as well, as in one of their "State of Robotics" videos on youtube, they discussed the future plans for expanding the voice recognition system with more languages. 
So, feel free to look them up on youtube, and stay tuned!
 
Cheers!

**Comment by moribundant** [comment]
Hello all, this is just a fantastic tool and it eclipses everything else out there...I just wish it would run natively on an M1 Mac 😉

As it is, I am running it with [Parallels](https://www.parallels.com) and performance is great. No issues so far and I have the EP Extension installed.
